component/src/main/java/org/wso2/extension/siddhi/io/report/generators/ReportGenerator.java

```diff
@@ -0,0 +1,151 @@
+                    break;
+                case "excel":
+                    fileName += ".xlsx";
+                    destFile = new File(fileName);
+                    JRXlsxExporter xlsxExporter = new JRXlsxExporter();
+                    xlsxExporter.setExporterInput(new SimpleExporterInput(jasperPrint));
+                    xlsxExporter.setExporterOutput(new SimpleOutputStreamExporterOutput(destFile));
+                    xlsxExporter.exportReport();
+                    break;
+                case "pdf":
+                    outputPath += ".pdf";
+                    JasperExportManager.exportReportToPdfFile(jasperPrint, outputPath);
+                    break;
+                default:
+                    break;
+
+            }
+        } catch (JRException e) {
+            throw new SiddhiAppRuntimeException("Failed to generate the " + reportProperties.get(ReportConstants
+                    .OUTPUT_FORMAT) + "file in " + outputPath, e);
+        }
+    }
+
+    public JasperPrint generateReportPrint(DynamicReport report, DynamicLayoutManager reportLayout,
+                                           JRBeanCollectionDataSource dataSource, Map<String, Object> parameters)
+            throws SiddhiAppRuntimeException {
+        JasperPrint jasperPrint;
+        try {
+            jasperPrint = DynamicJasperHelper.generateJasperPrint(report, reportLayout, dataSource,
+                    parameters);
+        } catch (JRException e) {
+            throw new SiddhiAppRuntimeException("Failed to generate the JasperPrint " + report.getReportName(), e);
+        } catch (ClassCastException e) {
+            throw new SiddhiAppRuntimeException("Failed to generate the report. Provide a numeric series column. ", e);
+        }
+        return jasperPrint;
+    }
+
+    public JasperPrint fillReportData(JasperReport jasperReport, Map<String, Object> parameters, JRDataSource
+            dataSource) {
+        JasperPrint jasperPrint;
+        try {
+            jasperPrint = JasperFillManager.fillReport(jasperReport, parameters, dataSource);
+        } catch (JRException e) {
+            throw new SiddhiAppRuntimeException("Failed to fill data into report template. " + jasperReport.getName()
+                    , e);
+        }
+        return jasperPrint;
+    }
+
+    public JasperDesign loadTemplate(String template) {
+        JasperDesign jasperDesign;
+        try {
+            jasperDesign = JRXmlLoader.load(template);
+        } catch (JRException e) {
+            throw new SiddhiAppRuntimeException("Failed to load the report template. " + template, e);
+        }
+        return jasperDesign;
+    }
+
+    public JasperReport compileTemplate(JasperDesign jasperDesign) {
+        JasperReport jasperReport;
+        try {
+            jasperReport = JasperCompileManager.compileReport(jasperDesign);
+        } catch (JRException e) {
+            throw new SiddhiAppRuntimeException("Failed to compile the report. ", e);
```

Comment:
What report? Is it possible to add more details to the exception message.

Reply:
Fixed with 096783872dcb45afcd00b778af443b4647ad246b